cluster/gce/gci/scaleoutserver-helper.sh

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Authors of Arktos - file created.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# A library of helper functions and constant for GCI distro
+source "${KUBE_ROOT}/cluster/gce/gci/helper.sh"
+
+# create-apiserver-instance creates the apiserver instance. If called with
+# an argument, the argument is used as the name to a reserved IP
+# address for the apiserver. (In the case of upgrade/repair, we re-use
+# the same IP.)
+#
+# It requires a whole slew of assumed variables, partially due to to
+# the call to write-apiservr-env. Listing them would be rather
+# futile. Instead, we list the required calls to ensure any additional
+#
+# variables are set:
+#   ensure-temp-dir
+#   detect-project
+#   get-bearer-token
+
+function create-proxy-instance {
+  local name=""
+  local address=""
+  local private_network_ip=""
+  [[ -n ${1:-} ]] && name="${1}"
+  [[ -n ${2:-} ]] && address="${2}"
+  [[ -n ${3:-} ]] && private_network_ip="${3}"
+  write-proxy-env
+  ensure-gci-metadata-files
+  create-proxy-instance-internal "${name}" "${address}" "${private_network_ip}"
+}
+
+function create-proxy-instance-internal() {
+  local gcloud="gcloud"
+  local retries=5
+  local sleep_sec=10
+  if [[ "${MASTER_SIZE##*-}" -ge 64 ]]; then  # remove everything up to last dash (inclusive)
+    # Workaround for #55777
+    retries=30
+    sleep_sec=60
+  fi
+
+  local -r server_name="${1}"
+  local -r address="${2:-}"
+  local -r private_network_ip="${3:-}"
+
+  local enable_ip_aliases
+  if [[ "${NODE_IPAM_MODE:-}" == "CloudAllocator" ]]; then
+    enable_ip_aliases=true
+  else
+    enable_ip_aliases=false
+  fi
+
+  local network=$(make-gcloud-network-argument \
+    "${NETWORK_PROJECT}" "${REGION}" "${NETWORK}" "${SUBNETWORK:-}" \
+    "${address:-}" "${private_network_ip:-}" "${enable_ip_aliases:-}" "${IP_ALIAS_SIZE:-}")
+
+  local metadata="kube-env=${KUBE_TEMP}/proxy-kube-env.yaml"
```

Comment:
this can be reduced by removing unneeded files:
1. proxy-kube-env.yaml (or rename just to be the proxy.yaml)
2. gci-ensure-docker-version.txt
3. gci-docker-version.txt
4. ${MASTER_EXTRA_METADATA}"

Reply:
1. renamed proxy-kube-env.yaml to proxy-env.yaml
2. gci-ensure-docker-version.txt/gci-docker-version.txt/${MASTER_EXTRA_METADATA}: commented this metadata for now and will reconsider if necessary when do TP/RP optimization